analyze: add "archs" verb that lists all known architectures

poettering opened the pull request:
<!-- devel-freezer = {"comment-id":"1814721130","freezing-tag":"v255-rc2"} -->

poettering commented:
This is a minor addition, but see no reason to sneak it into v255. Hence let's leave it for v256

bluca reviewed (commented):
Please add a call to this in TEST-65-ANALYZE - no need to verify its output, but just so that there's some coverage from sanitizer et al

yuwata commented on src/analyze/analyze-archs.c:
```diff
@@ -0,0 +1,89 @@
+                                return r;
+                }
+        else {
+                for (int i = 1; i < argc; i++) {
```
`STRV_FOREACH(x, strv_skip(argv, 1))`

yuwata commented on test/units/testsuite-65.sh:
```diff
@@ -866,6 +866,8 @@ systemd-analyze pcrs
 systemd-analyze pcrs --json=pretty
 systemd-analyze pcrs 14 7 0 ima
 
+systemd-analyze archs
```
Please also call with several names.

poettering commented:
Addressed all issues raised.

keszybz commented:
Hmm, shouldn't this support `--json`?

Also, is it "archs" or "arches"? "archs" looks strange and google says that it's not the correct spelling. That that is for "arch" in the sense of an architectural element, not our jargon abbreviation for "architecture". But we have "es" in "architectures", so I think we should here too.

poettering commented:
To avoid the archs vs. arches thing i renamed this to `architectures` now. No need to abbreviate here I think, it's not the most common of commands to execute I guess.

I also added the JSON output, and while at it also added support for it to "capability" and "exit-status" because trivial

bluca commented:
Please also add shell completion at least for bash

yuwata commented on src/analyze/analyze.c:
```diff
@@ -14,6 +14,7 @@
 #include "alloc-util.h"
 #include "analyze.h"
 #include "analyze-blame.h"
+#include "analyze-architectures.h"
```
Not sorted.

poettering replied:
fixed.

poettering commented:
> Please also add shell completion at least for bash

happy to review/merge a patch for that, but I don't think i am a bash guru enough to write that.

bluca commented:
> > Please also add shell completion at least for bash
> 
> happy to review/merge a patch for that, but I don't think i am a bash guru enough to write that.

It doesn't need to be fancy, just list the verb, simply add it to the STANDALONE list at  https://github.com/systemd/systemd/blob/main/shell-completion/bash/systemd-analyze#L65

poettering commented:
> It doesn't need to be fancy, just list the verb, simply add it to the STANDALONE list at https://github.com/systemd/systemd/blob/main/shell-completion/bash/systemd-analyze#L65

That doesn't look right nor helpful. The command does take parameters after all.
